olive/engine/packaging/packaging_generator.py

```diff
@@ -32,20 +40,124 @@
 
 
 def generate_output_artifacts(
-    packaging_config: PackagingConfig,
+    packaging_configs: Union[PackagingConfig, List[PackagingConfig]],
     footprints: Dict["AcceleratorSpec", "Footprint"],
     pf_footprints: Dict["AcceleratorSpec", "Footprint"],
     output_dir: Path,
+    azureml_client_config: "AzureMLClientConfig" = None,
 ):
     if sum(len(f.nodes) if f.nodes else 0 for f in pf_footprints.values()) == 0:
         logger.warning("No model is selected. Skip packaging output artifacts.")
         return
-    if packaging_config.type == PackagingType.Zipfile:
-        _generate_zipfile_output(packaging_config, footprints, pf_footprints, output_dir)
+    packaging_config_list = packaging_configs if isinstance(packaging_configs, list) else [packaging_configs]
+    for packaging_config in packaging_config_list:
+        if packaging_config.type == PackagingType.Zipfile:
+            _generate_zipfile_output(
+                packaging_config.name, packaging_config.config, footprints, pf_footprints, output_dir
+            )
+        elif packaging_config.type == PackagingType.AzureMLModels:
+            _generate_azureml_resources(
+                packaging_config.name,
+                packaging_config.config,
+                footprints,
+                pf_footprints,
+                azureml_client_config,
+                PackagingType.AzureMLModels,
+            )
+        elif packaging_config.type == PackagingType.AzureMLData:
+            _generate_azureml_resources(
+                packaging_config.name,
+                packaging_config.config,
+                footprints,
+                pf_footprints,
+                azureml_client_config,
+                PackagingType.AzureMLData,
+            )
+
+
+def _generate_azureml_resources(
+    output_name: str,
+    config: CommonPackagingConfig,
+    footprints: Dict["AcceleratorSpec", "Footprint"],
+    pf_footprints: Dict["AcceleratorSpec", "Footprint"],
+    azureml_client_config: "AzureMLClientConfig",
+    packaging_type: PackagingType,
+):
+    logger.info("Packaging output models to %s", packaging_type)
+
+    for accelerator_spec, pf_footprint in pf_footprints.items():
+        if pf_footprint.nodes and footprints[accelerator_spec].nodes:
+            model_rank = 1
```

Comment:
>model_rank [](http://example.com/codeflow?start=12&length=10)

does the model_rank json file need to be uploaded?

Reply:
No I don't think so. I prefer each asset no matter in Models or Data only contains this specific model data. But model rank file has other model data. But this reminds me we could split those data into different files and save to AML as well. Will update this.